e2e/actions/l2_challenger_test.go

```diff
@@ -56,6 +56,15 @@ func TestChallenger(gt *testing.T) {
 		AllowNonFinalized: false,
 	}, miner.EthClient(), challengerRollupClient)
 
+	guardianRPC := e2eutils.NewChallengerL2RPC(proposer.RPCClient())
+	guardianRollupClient := sources.NewRollupClient(guardianRPC)
+	guardian := NewL2Validator(t, log, &ValidatorCfg{
+		OutputOracleAddr:    sd.DeploymentsL1.L2OutputOracleProxy,
+		SecurityCouncilAddr: sd.DeploymentsL1.SecurityCouncilProxy,
+		ValidatorKey:        dp.Secrets.TrustedValidator,
```

Comment:
The component `validator` with malicious rpc also has the address of trusted validator, how about change this guardian's `ValidatorKey` to `dp.Secrets.Challenger` as in `e2e/setup.go`?
I think it's weird the malicious validator and guardian have the same `ValidatorKey` even though it's test case.